pytorch_lightning/utilities/data.py

```diff
@@ -18,28 +18,55 @@
 from torch.utils.data import DataLoader, IterableDataset
 
 from pytorch_lightning.utilities import rank_zero_warn
+from pytorch_lightning.utilities.warnings import WarningCache
 
 BType = Union[torch.Tensor, str, Mapping[Any, "BType"], Iterable["BType"]]
 
+warning_cache = WarningCache()
+
+
+def _extract_batch_size(batch: BType, batch_sizes) -> bool:
+    current_batch_size = None
+
+    if isinstance(batch, torch.Tensor):
+        current_batch_size = batch.size(0)
+    elif isinstance(batch, str):
+        current_batch_size = len(batch)
+    elif not isinstance(batch, (Iterable, dict)):
```

Comment:
@rohitgr7 
Instead of using `dict`, `Mapping` should be used.
Example for a case in which the batch will be `Mapping` but not `dict` is when using huggingface.
Currently, 9 is returned instead of 2 for this HF example code:
```
from transformers import AutoTokenizer
tokenizer = AutoTokenizer.from_pretrained("bert-base-uncased")
pt_batch = tokenizer(
    ["We are very happy to show you the 🤗 Transformers library.", "We hope you don't hate it."],
    padding=True,
    truncation=True,
    max_length=512,
    return_tensors="pt"
)
print(f"Inferred batch size is {extract_batch_size(pt_batch)}")
```

Reply:
true.. it's a `UserDict`.. thanks you :)